packages/marshal/src/encodePassable.js

```diff
@@ -325,14 +509,51 @@ const decodeTagged = (encoded, decodePassable) => {
  */
 export const makeEncodePassable = (encodeOptions = {}) => {
   const {
-    encodeRemotable = (rem, _) => Fail`remotable unexpected: ${rem}`,
-    encodePromise = (prom, _) => Fail`promise unexpected: ${prom}`,
-    encodeError = (err, _) => Fail`error unexpected: ${err}`,
+    encodeRemotable: unsafeEncodeRemotable = (r, _) =>
+      Fail`remotable unexpected: ${r}`,
+    encodePromise: unsafeEncodePromise = (p, _) =>
+      Fail`promise unexpected: ${p}`,
+    encodeError: unsafeEncodeError = (err, _) => Fail`error unexpected: ${err}`,
+    format = 'legacyOrdered',
   } = encodeOptions;
 
-  const encodePassable = passable => {
+  let formatPrefix;
+  let encodeStringSuffix;
+  let encodeArray;
+  if (format === 'legacyOrdered') {
+    formatPrefix = '';
+    encodeStringSuffix = encodeLegacyStringSuffix;
+    encodeArray = encodeLegacyArray;
+  } else if (format === 'compactOrdered') {
+    formatPrefix = '~';
+    encodeStringSuffix = encodeCompactStringSuffix;
+    encodeArray = encodeCompactArray;
+  } else {
+    Fail`Unrecognized format: ${q(format)}`;
+  }
+  const encodeRemotable = makeEncodeRemotable(
+    unsafeEncodeRemotable,
+    encodeStringSuffix,
+  );
+  const encodePromise = makeEncodePromise(
+    unsafeEncodePromise,
+    encodeStringSuffix,
+  );
+  const encodeError = makeEncodeError(unsafeEncodeError, encodeStringSuffix);
```

Comment:
This double-escapes the contents of any embedded string (such as for the error message), but I don't see a good alternative. In particular, asserting that `unsafeEncodeError` produced an independently valid encoding that needs no further escaping would be awkward because this code doesn't have access to any caller-provided decode functions for embedded remotables/promises/errors.